pkg/register/restResolver.go

```diff
@@ -51,11 +86,25 @@ func NewRestResolverClient(url *url.URL, timeout time.Duration) ResolverClient {
 	)
 }
 
-// NewRestResolverClientWithCustomClient build a ResolverClient with customer http.Client
+// NewRestResolverClientWithCustomClient build a ResolverClient with custom http.Client
 func NewRestResolverClientWithCustomClient(url *url.URL, client *http.Client) ResolverClient {
-	return &RestResolverClient{
-		url:    url,
-		client: client,
+	return &MeteredRestResolverClient{
+		RestResolverClient: &RestResolverClient{
+			url:    url,
+			client: client,
+		},
+		meter: &NoopResolverClientMetrics{},
+	}
+}
+
+// NewRestResolverClientWithMeteredClient build a ResolverClient with custom http.Client & metrics
+func NewRestResolverClientWithMeteredClient(url *url.URL, client *http.Client, meter ResolverClientMetrics) ResolverClient {
+	return &MeteredRestResolverClient{
+		RestResolverClient: &RestResolverClient{
+			url:    url,
+			client: client,
+		},
+		meter: meter,
```

Comment:
check whether meter is nil (otherwise will have nil dereferences later) & return error? (or - blindly switch to noop?)